python/felis/datamodel.py

```diff
@@ -942,3 +986,118 @@ def __contains__(self, id: str) -> bool:
             The ID of the object to check.
         """
         return id in self.id_map
+
+    def find_object_by_id(self, id: str, obj_type: type[T] = BaseObject) -> T:  # type: ignore
```

Comment:
If you don't want to ignore, perhaps casting? (Or not setting a default here) might make things easier: 
```
    def find_object_by_id(self, id: str, obj_type: Type[T] = cast(Type[T], BaseObject)) -> T:
```
But then you also need Schema to be a Generic:

```
class Schema(BaseObject, Generic[T]):
```